AK/windows.h

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2024, stasoid <stasoid@yahoo.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#ifdef AK_OS_WINDOWS // needed for Swift
+#    define timeval dummy_timeval
+#    include <winsock2.h>
+#    undef timeval
+#    include <io.h>
+#endif
```

Comment:
idea: be more restrictive with ideas like in the following?

```suggestion
#ifndef AK_OS_WINDOWS
	static_assert(false, "Windows header included on non-Windows target");
#endif

// needed for Swift
#define timeval dummy_timeval
#include <winsock2.h>
#undef timeval

#include <io.h>
```